packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -84,6 +84,14 @@ function Unstable_TrapFocus(props) {
         return;
       }
 
+      // Support focused nodes removed from DOM with an interval call.
+      if (!event) {
+        if (doc.activeElement.tagName === 'BODY') {
+          rootRef.current.focus();
+        }
+        return;
+      }
```

Comment:
I have tried to limit the side effects of the interval to the minimum required. It should help in #21610. cc @eps1lon.

Reply:
Wondering if we should just not re-use contain for the interval. Since we'll be returning early and the earlier check is the only common code?

Reply:
It seems that the branch logic before this one has to be shared.

Reply:
We should probably add a test to cover this fact.